common/network-common/src/main/java/org/apache/spark/network/util/LevelDBProvider.java

```diff
@@ -80,7 +80,12 @@ public static DB initLevelDB(File dbFile, StoreVersion version, ObjectMapper map
         }
       }
       // if there is a version mismatch, we throw an exception, which means the service is unusable
-      checkVersion(tmpDb, version, mapper);
+      try {
+        checkVersion(tmpDb, version, mapper);
+      } catch (IOException ioe) {
```

Comment:
https://github.com/apache/spark/blob/83434af22284482d4e805d8be625f733b074928b/common/network-common/src/main/java/org/apache/spark/network/util/LevelDBProvider.java#L110-L111

For LevelDB, it will only throw IOException.